### PR TITLE
Add ExPlat implementation to :experimentation module

### LINKED ISSUE #112: Extract ExPlat from WordPress-Android to Automattic-Tracks-Android
Extract [ExPlat implementation](https://github.com/wordpress-mobile/WordPress-Android/tree/trunk/WordPress/src/main/java/org/wordpress/android/util/experiments) from WordPress-Android to Automattic-Tracks-Android so it can also be used by [woocommerce-android](https://github.com/woocommerce/woocommerce-android) and other projects.

### PR DESCRIPTION
**Pull request by RenanLukas**
Resolves #112

Copies ExPlat implementation from `WordPress-Android` so it can be used by other projects that depend on `Automattic-Tracks-Android`.

### CODE REVIEW

File: experimentation/src/main/java/com/automattic/android/experimentation/ExPlat.kt [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import org.wordpress.android.util.AppLog.T
+
+class ExPlat(
+    private val experiments: Set<Experiment>,
+    private val experimentStore: ExperimentStore,
+    private val appLogWrapper: AppLogWrapper,
+    private val coroutineScope: CoroutineScope,
+    private val isDebug: Boolean
+) {
+    private val platform = Platform.WORDPRESS_ANDROID
```
Comment (wzieba) [inline experimentation/src/main/java/com/automattic/android/experimentation/ExPlat.kt]: I believe this should be declared by client

File: experimentation/src/main/java/com/automattic/android/experimentation/example/ExampleExperiment.kt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.automattic.android.experimentation.example
+
+import com.automattic.android.experimentation.ExPlat
+import com.automattic.android.experimentation.Experiment
+
+@Suppress("unused")
+class ExampleExperiment(exPlat: ExPlat) : Experiment(name = "example_experiment", exPlat = exPlat)
```
Comment (wzieba) [inline experimentation/src/main/java/com/automattic/android/experimentation/example/ExampleExperiment.kt]: If it's not used anywhere, maybe we can remove it to not cause confusion?

File: experimentation/src/test/java/com/automattic/android/experimentation/ExPlatTest.kt [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+import org.wordpress.android.fluxc.model.experiments.Variation
+import org.wordpress.android.fluxc.model.experiments.Variation.Control
+import org.wordpress.android.fluxc.model.experiments.Variation.Treatment
+import org.wordpress.android.fluxc.store.ExperimentStore
+import org.wordpress.android.fluxc.store.ExperimentStore.OnAssignmentsFetched
+import org.wordpress.android.fluxc.utils.AppLogWrapper
+import java.util.Date
+
+@ExperimentalCoroutinesApi
+class ExPlatTest {
```
Comment (wzieba) [inline experimentation/src/test/java/com/automattic/android/experimentation/ExPlatTest.kt:29]: Thanks, love the coverage!

File: experimentation/build.gradle [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        versionName "1.0"
+
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+    }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    lintOptions {
+        baseline file("lint-baseline.xml")
```
Comment (wzieba) [inline experimentation/build.gradle]: As it's a new module, the baseline is not needed

File: experimentation/build.gradle [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+android {
+    compileSdkVersion 32
+
+    defaultConfig {
+        minSdkVersion 21
+        targetSdkVersion 30
+        versionCode 1
+        versionName "1.0"
+
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
```
Comment (wzieba) [inline experimentation/build.gradle]: As this module has no instrumentation tests, let's remove this to not add anything not needed.

File: experimentation/src/main/java/com/automattic/android/experimentation/ExPlat.kt (hunk 1 above)
Comment (RenanLukas) [reply experimentation/src/main/java/com/automattic/android/experimentation/ExPlat.kt]: Great catch! Thank you

File: experimentation/src/test/java/com/automattic/android/experimentation/ExPlatTest.kt (hunk 3 above)
Comment (RenanLukas) [reply experimentation/src/test/java/com/automattic/android/experimentation/ExPlatTest.kt:29]: Thank you for implementing it @renanferrari 😁

**Comment by RenanLukas** [comment]
Thank you for the initial review @wzieba ! I've addressed your comments.

Any ideas on why CI is complaining about `:AutomatticTracks`?

**Comment by wzieba** [comment]
@RenanLukas I'm investigating this issue, for this moment:
- lint started to fail on `Cursor#getColumnIndex`
- the quick fix is a change: `Cursor#getColumnIndex` -> `Cursor#getColumnIndexOrThrow`
- this happens in different projects as well (e.g. https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2295/commits/979125fce552767b179f3c01611615a64d21b237)
- the weird thing is that it happens also if you checkout some older version of the library (e.g. commit `a957ae9`, when it was passing few months ago when it was building on CI). The same thing is reported by [different people](https://stackoverflow.com/questions/69053061/android-studio-value-must-be-%E2%89%A5-0#comment122160428_69053061). I'm wondering what is the case, because version of `Android Lint` is associated with `Android Gradle Plugin` so without any code changes it shouldn't break 🤔 . 

Either way this issue is not directly associated with your code changes. I believe it'd be awesome to fix it in a separate PR but ideally it'd be awesome to find a root cause - `EventTable` shouldn't be touched by any means, nothing really changed in implementation.

File: experimentation/build.gradle [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        content {
+            includeGroup "org.wordpress"
+            includeGroup "org.wordpress.fluxc"
+            includeGroup "org.wordpress.wellsql"
+        }
+    }
+}
+
+android {
+    compileSdkVersion 32
```
Comment (wzieba) [inline experimentation/build.gradle]: @RenanLukas generally, the core issue is that `compileSdkVersion` of this module, influenced which version of lint/lint rules will be used. So lowering it down to `30` will resolve the issue. I think the best will be to bump `compileSdkVersion` of both modules in a seperate PR.

Here is branch on which I was testing, see that build is passing on CI: https://github.com/Automattic/Automattic-Tracks-Android/compare/issue/112-extract-explat...issue/112-testing?expand=1

Comment (RenanLukas) [reply experimentation/build.gradle]: Thank you very much @wzieba 👍

**Review by renanferrari — APPROVED** [review]
Thanks for handling this! 

LGTM 🚢

**Review by wzieba — APPROVED** [review]
Looks good to me!

**Pull request merged**